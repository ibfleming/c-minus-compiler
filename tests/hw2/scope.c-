// C-F24
int g;

main() {
    int a;
    {
        int b;
        {
            int c;
        }
    }
    {
        int d;
        {
            int e;
            int f;
        }
    }
}
