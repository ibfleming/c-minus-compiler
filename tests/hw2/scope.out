Var: g  [line: 2]
Sibling: 1  Func: main  [line: 4]
.   Child: 1  Compound [line: 4]
.   .   Child: 0  Var: a  [line: 5]
.   .   Child: 1  Compound [line: 6]
.   .   .   Child: 0  Var: b  [line: 7]
.   .   .   Child: 1  Compound [line: 8]
.   .   .   .   Child: 0  Var: c  [line: 9]
.   .   Sibling: 1  Compound [line: 12]
.   .   .   Child: 0  Var: d  [line: 13]
.   .   .   Child: 1  Compound [line: 14]
.   .   .   .   Child: 0  Var: e  [line: 15]
.   .   .   .   Sibling: 1  Var: f  [line: 16]
